src/FluentAssertions.Analyzers/Tips/Collections/CollectionAnalyzer.cs

```diff
@@ -1,20 +1,252 @@
-﻿using FluentAssertions.Analyzers.Utilities;
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using FluentAssertions.Analyzers.Utilities;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace FluentAssertions.Analyzers;
 
-public abstract class CollectionAnalyzer : FluentAssertionsAnalyzer
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class CollectionAnalyzer : FluentAssertionsAnalyzer
```

Comment:
introduce a base class for collections to make the non-migrated analyzers not extend this class